### PR TITLE
Fixes for rxe_cfg

### PR DESCRIPTION
**Pull request by jgunthorpe**
Based on some light testing of the Debian packaging.

@bdrung @monis410

### CODE REVIEW

**Comment by rleon** [comment]
@jgunthorpe I'm fine with rxe patches.

File: providers/rxe/rxe_cfg [hunk 1]
```diff
@@ -650,7 +650,7 @@ sub main {
     }
 
     # create persistence file if necessary
-    mkdir -p $persistence_path;
+    system("mkdir -p $persistence_path");
```
Comment (bdrung) [inline providers/rxe/rxe_cfg]: Please do not use the `system` call. Instead use `make_path` which is the corresponding Perl function.

File: debian/control [hunk 2]
```diff
@@ -23,6 +23,7 @@ Homepage: https://github.com/linux-rdma/rdma-core
 Package: rdma-core
 Architecture: linux-any
 Depends: dmidecode, ${misc:Depends}, ${shlibs:Depends}
+Recommends: dmidecode, ethtool, net-tools
```
Comment (bdrung) [inline debian/control]: You forgot to remove dmidecode from Depends. Please put the move of dmidecode in a separate commit.

File: providers/rxe/rxe_cfg (hunk 1 above)
Comment (jgunthorpe) [inline providers/rxe/rxe_cfg]: Used this:
```
--- a/providers/rxe/rxe_cfg.in
+++ b/providers/rxe/rxe_cfg.in
@@ -36,6 +36,7 @@ use warnings;
 use strict;
 
 use File::Basename;
+use File::Path qw(make_path);
 use Getopt::Long;
 
 my $help = 0;
@@ -650,7 +651,7 @@ sub main {
     }
 
     # create persistence file if necessary
-    system("mkdir -p $persistence_path");
+    make_path($persistence_path);
     if (!(-e $persistence_file)) {
         `touch $persistence_file`;
     }
```

File: debian/control (hunk 2 above)
Comment (jgunthorpe) [inline debian/control]: Done

**Comment by jgunthorpe** [comment]
Updated with bdrung's two comments
